### PR TITLE
Adjust required permissions for system schema

### PR DESCRIPTION
**Pull request by jon-wei**
This PR implements the following change described in #7563:

> The permissions needed for system table access could be adjusted to not require additional permissions beyond datasource-specific permissions for non-server related info (segments, tasks) and to require STATE-read permissions for server-related tables (servers, server_segments) to be consistent with the non-SQL APIs.

A STATE.READ permission check is added to `server_segments` sys table handling (`servers` already had one, and #7567 removed the unnecessary `sys` datasource check).

This PR also adds integration tests for sys schema + auth. A new segment and task for `auth_test` datasource are inserted directly into metadata during IT setup, and this datasource is used for testing the auth checks.

### CODE REVIEW

**Comment by jon-wei** [comment]
will need to adjust tests to handle when the full IT test suite is run (since cluster state is not isolated across tests, things like "total stored segment size on historicals" change)

File: sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java [hunk 1]
```diff
@@ -493,11 +493,30 @@ public TableType getJdbcTableType()
...
+      final AuthenticationResult authenticationResult =
+          (AuthenticationResult) root.get(PlannerContext.DATA_CTX_AUTHENTICATION_RESULT);
+
+      final Access stateAccess = AuthorizationUtils.authorizeAllResourceActions(
+          authenticationResult,
+          Collections.singletonList(new ResourceAction(new Resource("STATE", ResourceType.STATE), Action.READ)),
+          authorizerMapper
+      );
+      if (!stateAccess.isAllowed()) {
+        throw new ForbiddenException("Insufficient permission to view servers :" + stateAccess);
```
[the remark this answers is not in the document]
Comment (jon-wei) [reply sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java]: Fixed

File: sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java [hunk 2]
```diff
@@ -493,11 +493,30 @@ public TableType getJdbcTableType()
     @Override
     public Enumerable<Object[]> scan(DataContext root)
     {
+      final AuthenticationResult authenticationResult =
+          (AuthenticationResult) root.get(PlannerContext.DATA_CTX_AUTHENTICATION_RESULT);
+
+      final Access stateAccess = AuthorizationUtils.authorizeAllResourceActions(
+          authenticationResult,
+          Collections.singletonList(new ResourceAction(new Resource("STATE", ResourceType.STATE), Action.READ)),
```
[the remark this answers is not in the document]
Comment (jon-wei) [reply sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java]: Made this a constant in Resource

File: sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java [hunk 3]
```diff
@@ -493,11 +493,30 @@ public TableType getJdbcTableType()
...
+          authorizerMapper
+      );
```
[the remark this answers is not in the document]
Comment (jon-wei) [reply sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java]: Moved this to a checkStateReadAccessForServers method

**Comment by jon-wei** [comment]
I updated the SQL docs with info on necessary permissions, also added tests for a user without access to the `auth_test` datasource

File: docs/content/development/extensions-core/druid-basic-security.md [hunk 4]
```diff
@@ -310,6 +310,18 @@ For information on what HTTP methods are supported on a particular request endpo
 
 GET requires READ permission, while POST and DELETE require WRITE permission.
 
+### SQL Permissions
+
+Queries on Druid datasources require DATASOURCE READ permissions for the specified datasource.
+
+Queries on the [information schema tables](../../querying/sql.html#information-schema) require DATASOURCE READ access for the specified datasource.
```
Comment (gianm) [inline docs/content/development/extensions-core/druid-basic-security.md]: Please adjust the language a bit- there is not necessarily any 'specified datasource' with the information schema. The query might just be `SELECT * FROM INFORMATION_SCHEMA.COLUMNS`. I'd suggest:

```
Queries on the [INFORMATION_SCHEMA tables](../../querying/sql.html#information-schema) will
return information about datasources that the caller has DATASOURCE READ access to. Other
datasources will be omitted.
```

**Comment by gianm** [comment]
LGTM other than the comment about doc wording.

File: docs/content/development/extensions-core/druid-basic-security.md (hunk 4 above)
Comment (jon-wei) [reply docs/content/development/extensions-core/druid-basic-security.md]: Thanks, reworded

**Comment by jon-wei** [comment]
Fixed tests:
- Updated to handle changes from is_overshadowed patch
- Changed the created year for the inserted task to 2030 so it doesn't fall out of the completed tasks window

**Pull request merged**